py-polars/tests/unit/functions/as_datatype/test_array.py

```diff
@@ -0,0 +1,77 @@
+import polars as pl
+from polars.testing import assert_series_equal
+
+
+def test_array() -> None:
```

Comment:
Would eventually love tests for
- an empty list with no datatype errors
- an empty list with a datatype works

Reply:
Sounds good, I need to do some investigation on whether polars supports 0 length arrays since this is what I would expect out of an empty list.

Reply:
@NickCrews so, from what I can tell, there does not seem to be the concept of an empty Array in polars. That is to say, the rows in an array must have a positive length.
So, for example, we can have empty lists:
```
s = pl.Series("A", [[], []], dtype = pl.List(pl.Int64))
print(s)

shape: (2,)
Series: 'A' [list[i64]]
[
	[]
	[]
]
```

If we try to convert this to an array we get an error:
```
a = s.list.to_array(0)
```
Gives the error: `polars.exceptions.ComputeError: not all elements have the specified width 0`

Similarly, we can't directly construct an empty array. It seems the closest we can get is:
```
s = pl.Series("A", [[None], [None]], dtype = pl.Array(pl.Int64, 1))
print(s)

shape: (2,)
Series: 'A' [array[i64, 1]]
[
	[null]
	[null]
]
```

For now, if we call this function with an empty array like 
`df.select(pl.array([], dtype='{"DtypeColumn":["Float64"]}').alias("z"))["z"]` 
what we get is 
`polars.exceptions.ComputeError: 'array' needs one or more expressions`

At any rate, right now it is not clear what we should return for an empty array other than an error.

Reply:
The empty array was added in https://github.com/pola-rs/polars/pull/18940 (1.9.0 release)

```python
s.list.to_array(0)
# shape: (2,)
# Series: 'A' [array[i64, 0]]
# [
# 	[]
# 	[]
# ]

```